src/sync_client.rs

```diff
@@ -256,31 +263,93 @@ pub async fn run(
 	pp: PublicParameters,
 	block_tx: Option<Sender<BlockVerified>>,
 ) {
+	let rpc_client = sync_client.get_client();
 	if sync_blocks_depth >= 250 {
 		warn!("In order to process {sync_blocks_depth} blocks behind latest block, connected nodes needs to be archive nodes!");
 	}
+
+	// Fetch validator set at current height
+	let final_validator_set = rpc::get_valset_by_block_number(&rpc_client, end_block)
+		.await
+		.unwrap();
```

Comment:
Should we panic inside the sync client or handle these errors differently?

Reply:
Yes, I will modify sync client to act more like LC, with better error handling. I meant to do that anyway, but forgot.

Reply:
At the second look, I think we should panic. Whereas we can have errors processing blocks themselves, without failing, errors in checking the validity should be fatal. I will of course add some more context, but I think it should stay.